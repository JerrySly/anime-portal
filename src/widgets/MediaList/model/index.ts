import { Ref, ref } from "vue";

export const loaded: Ref<boolean> = ref(false);