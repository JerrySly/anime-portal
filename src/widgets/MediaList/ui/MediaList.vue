<template>
  <div v-if="items?.length">
    <h3 class="title">{{ title }}</h3>
    <vue-horizontal responsive v-if="items?.length" class="horizontal">
      <media-list-item
        v-for="(item, index) in items"
        class="media-list-item"
        :key="index"
        :id="item?.id"
        :img="item?.coverImage"
        :title="item?.title"
      />
    </vue-horizontal>
  </div>
  <div class="skeleton-list" v-else>
    <div class="skeleton-list__title-block">
      <v-skeleton-loader
          class="skeleton-list__title"
          type="subtitle"
          width="700"
      />
    </div>
    <vue-horizontal responsive>
      <v-skeleton-loader
        v-for="(_, index) in 10"
        class="skeleton-list__item"
        :key="index"
        type="card"
        width="200"
        min-height="260"
      />
    </vue-horizontal>
  </div>
</template>

<script lang="ts">
import VueHorizontal from 'vue-horizontal';
import MediaListItem from '@/entities/media/ui/MediaListItem.vue';
import { PropType, watch } from 'vue';
import { Media } from '@/shared/__generated__/graphql';
import { loaded } from '@/pages/feed/model';

export default {
  name: 'MediaList',
  components: { 
    VueHorizontal,
     MediaListItem },
  props: {
    items: {
      type: Object as PropType<Array<Media>>,
      require: true,
    },
    title: {
      type: String,
      require: true,
    }
  },
  setup(props: any) { // TODO: Нужно поставить нужный тип
    return {
      ...props,
      loaded,
    }
  },
}
</script>

<style lang="scss" scoped>
.horizontal {
  :deep(.v-hl-btn-next) {
    top: 120px !important;
    bottom: 0 !important;
    right: -20px !important;
    transform: translateX(0) !important;
    align-self: flex-start !important;
  }
  :deep(.v-hl-btn-prev) {
    top: 120px !important;
    bottom: 0 !important;
    left: -15px !important;
    transform: translateX(0) !important;
    align-self: flex-start !important;
  }
}
.skeleton-list {
  &__title-block {
    display: flex;
    justify-content: center;
  }
  &__title {

  }
  &__item {
    margin: 0px 10px 0px 10px;
  }
}
.title {
  font-size: 28px;
}
.media-list-item {
  padding: 5px 10px;
}
</style>