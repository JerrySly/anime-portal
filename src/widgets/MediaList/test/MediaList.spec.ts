import { mount, shallowMount } from "@vue/test-utils";
import MediaList from "../ui/MediaList.vue";
import { ApolloClients   } from '@vue/apollo-composable'
import { Media } from "@/shared/__generated__/graphql";
import VueHorizontal from "vue-horizontal";

const media: Media = {
  id: 1,
  isFavourite: false,
  isFavouriteBlocked: false,
  coverImage: {
    large: 'test',
  }
} 

test('Render title', async () => {

  const customTitle = 'CUSTOM TITLE';

  const item = mount(MediaList, {
    props: {
      title: customTitle,
      items: [
        media
      ]
    }
  })
  await item.vm.$nextTick();
  console.log(item.html());
  expect(item.find('.title').text()).toEqual(customTitle);
})

test('Render carousel', () => {
  
  const item = mount(MediaList, {
    props: {
      title: '',
      items: [
        media
      ]
    }
  })

  expect(item.getComponent(VueHorizontal)).toMatchSnapshot();
})