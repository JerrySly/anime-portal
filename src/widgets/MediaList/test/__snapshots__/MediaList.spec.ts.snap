// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render carousel 1`] = `
VueWrapper {
  "__app": null,
  "__setProps": undefined,
  "cleanUpCallbacks": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": Object {
    "button": true,
    "buttonBetween": true,
    "calculate": [Function],
    "children": [Function],
    "debounceId": undefined,
    "displacement": 1,
    "findNextSlot": [Function],
    "findPrevSlot": [Function],
    "hasNext": false,
    "hasPrev": false,
    "left": 0,
    "next": [Function],
    "onScroll": [Function],
    "onScrollDebounce": [Function],
    "prev": [Function],
    "refresh": [Function],
    "scroll": false,
    "scrollToIndex": [Function],
    "scrollToLeft": [Function],
    "scrollWidth": 0,
    "snap": "start",
    "width": 0,
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "items": Array [
        Object {
          "coverImage": Object {
            "large": "test",
          },
          "id": 1,
          "isFavourite": false,
          "isFavouriteBlocked": false,
        },
      ],
      "loaded": false,
      "title": "",
    },
  },
  "wrapperElement": <div
    class="vue-horizontal"
    responsive=""
    style="position: relative; display: flex;"
  >
    <!---->
    <!---->
    <div
      class="v-hl-container"
      style="display: flex; width: 100%; margin: 0px 0px -30px 0px; padding: 0px 0px 30px 0px; box-sizing: content-box; overflow-x: scroll; overflow-y: hidden; scroll-snap-type: x mandatory; clip-path: inset(0 0 30px 0);"
    >
      
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      <v-skeleton-loader
        class="skeleton-list__item"
        min-height="260"
        style="flex-shrink: 0; box-sizing: border-box; min-height: 1px; scroll-snap-align: start;"
        type="card"
        width="200"
      />
      
    </div>
  </div>,
}
`;
